packages/forms/resources/js/components/file-upload.js

```diff
@@ -179,6 +179,8 @@ export default (Alpine) => {
                                 file.startsWith('livewire-file:'),
                             ).length
                         ) {
+                            this.lastState = null
```

Comment:
Unfortunately this didnt fix the problem for me? Try the Create Product page in the demo.

Reply:
Weird works on the create blog post page for me.

Reply:
What would be different about those?

Reply:
Can you confirm that it doesnt work for products though or is that just me?

Reply:
It doesn't work on the products page for me, while it does work on the blog post page.

Reply:
I'll see if I can figure it out today.

Reply:
So, this is something to do with the product page / resource and i'm not sure what it could be.

https://user-images.githubusercontent.com/3596800/195993445-63062d96-b75d-4f71-ac54-539819613d3b.mov

Reply:
Oddly enough, without the fix. The Spatie upload clears correctly.

Reply:
Latest commit addresses the issue on the product page. For some reason on that resource the media field is coming back as null after saving which the JS can't convert to an object to perform the necessary check for if the file list has changed or not.